### ISSUE TITLE
Feature request: rules_shell should support hermetic toolchain

### ISSUE THREAD

**Issue by BoleynSu** [bazelbuild/rules_shell#4]

**Comment by fmeum** [comment]
Yes, but getting it right requires some thought and design work:
* rules_shell so far only provides a shell *runtime* toolchain. Making that hermetic is nice, but I would consider introducing a hermetic *exec* toolchain (for `ctx.actions.run_shell`) to be more impactful. It's not completely clear how to wire this up in Java though.
* On non-Windows platforms, shell scripts are executed directly without any kind of launcher. A hermetic toolchain would probably require introducing one as shebangs must be absolute and have length restrictions.

**Comment by EdSchouten** [comment]
Can't we deprecate `ctx.actions.run_shell()` altogether?

**Comment by fmeum** [comment]
> Can't we deprecate `ctx.actions.run_shell()` altogether?

That is an appealing option, but we would need a well designed replacement that makes it easy to supply hermetic tools. `ape` and `rules_sh` can provide inspiration.
